fastestimator/trace/metric/auc.py

```diff
@@ -0,0 +1,153 @@
+        mode: What mode(s) to execute this Trace in. For example, "train", "eval", "test", or "infer". To execute
+            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
+            like "!infer" or "!train".
+        ds_id: What dataset id(s) to execute this Trace in. To execute regardless of ds_id, pass None. To execute in all
+            ds_ids except for a particular one, you can pass an argument like "!ds1".
+        output_name: Name of the key to store back to the state.
+        average: {'micro', 'macro', 'samples', 'weighted'} or None,
+            If ``None``, the scores for each class are returned. Otherwise, this determines the
+            type of averaging performed on the data. Note: multiclass ROC AUC currently only handles the 'macro' and
+            'weighted' averages. For multiclass targets, `average=None` is only implemented for `multi_class='ovo'`.
+            ``'micro'``:
+                Calculate metrics globally by considering each element of the label
+                indicator matrix as a label.
+            ``'macro'``:
+                Calculate metrics for each label, and find their unweighted
+                mean.  This does not take label imbalance into account.
+            ``'weighted'``:
+                Calculate metrics for each label, and find their average, weighted
+                by support (the number of true instances for each label).
+            ``'samples'``:
+                Calculate metrics for each instance, and find their average.
+            Will be ignored when ``y_true`` is binary.
+        sample_weight: array-like of shape (n_samples,), default=None. Sample weights.
+        max_fpr: float > 0 and <= 1, default=None
+            If not ``None``, the standardized partial AUC over the range [0, max_fpr] is returned.
+            For the multiclass case, ``max_fpr``, should be either equal to ``None`` or ``1.0`` as AUC ROC partial
+            computation currently is not supported for multiclass.
+        multi_class : {'raise', 'ovr', 'ovo'}, default='raise'
+            Only used for multiclass targets. Determines the type of configuration to use. The default value raises
+            an error, so either
+            ``'ovr'`` or ``'ovo'`` must be passed explicitly.
+            ``'ovr'``:
+                Stands for One-vs-rest. Computes the AUC of each class against the rest. This
+                treats the multiclass case in the same way as the multilabel case. Sensitive to class imbalance even
+                when ``average == 'macro'``, because class imbalance affects the composition of each of the
+                'rest' groupings.
+            ``'ovo'``:
+                Stands for One-vs-one. Computes the average AUC of all possible pairwise combinations of classes.
+                Insensitive to class imbalance when ``average == 'macro'``.
+        labels : array-like of shape (n_classes,), default=None
```

Comment:
what is the actual value/purpose of this argument? There seem to be pretty significant limitations on what it can be set to (https://github.com/scikit-learn/scikit-learn/blob/6cb2c5237/sklearn/metrics/_ranking.py#L365), and I doubt that the type-hint of float will be correct for most cases.